src/serializers/encoder/v11/encode_schema.c

```diff
@@ -49,7 +49,18 @@ static inline void _RdbSaveIndexData(RedisModuleIO *rdb, Index *idx) {
 	RedisModule_SaveUnsigned(rdb, fields_count);
 	for(uint i = 0; i < fields_count; i++) {
 		// Indexed property
-		RedisModule_SaveStringBuffer(rdb, idx->fields[i], strlen(idx->fields[i]) + 1);
+		RedisModule_SaveStringBuffer(rdb, idx->fields[i].name, strlen(idx->fields[i].name) + 1);
+		if(idx->type == IDX_FULLTEXT) {
```

Comment:
I think `_RdbSaveIndexData` should further call `_RdbSaveExactMatchIndex` or `_RdbSaveFullTextIndexData` depending on the type of index being encoded, same thing for the decoder counter part